packages/cells/src/cellDragUtils.ts

```diff
@@ -0,0 +1,209 @@
+/*-----------------------------------------------------------------------------
+| Copyright (c) Jupyter Development Team.
+| Distributed under the terms of the Modified BSD License.
+|----------------------------------------------------------------------------*/
+
+/**
+ * This module contains some utility functions to operate on cells. This
+ * could be shared by widgets that contain cells, like the CodeConsole or
+ * Notebook widgets.
+ */
+
+import { each, IterableOrArrayLike } from '@phosphor/algorithm';
+import { ICodeCellModel } from './model';
+import { Cell } from './widget';
+import { h, VirtualDOM } from '@phosphor/virtualdom';
+import { nbformat } from '../../coreutils/lib';
+
+
+/**
+ * Constants for drag 
+ */
+
+/**
+ * The threshold in pixels to start a drag event.
+ */
+const DRAG_THRESHOLD = 5;
+/**
+ * The class name added to drag images.
+ */
+const DRAG_IMAGE_CLASS = 'jp-dragImage';
+
+/**
+ * The class name added to singular drag images
+ */
+const SINGLE_DRAG_IMAGE_CLASS = 'jp-dragImage-singlePrompt';
+
+/**
+ * The class name added to the drag image cell content.
+ */
+const CELL_DRAG_CONTENT_CLASS = 'jp-dragImage-content';
+
+/**
+ * The class name added to the drag image cell content.
+ */
+const CELL_DRAG_PROMPT_CLASS = 'jp-dragImage-prompt';
+
+/**
+ * The class name added to the drag image cell content.
+ */
+const CELL_DRAG_MULTIPLE_BACK = 'jp-dragImage-multipleBack';
+
+
+/**
+ * Find the cell index containing the target html element.
+ * This function traces up the DOM hierarchy to find the root cell
+ * node. Then find the corresponding child and select it.
+ *
+ * @param node - the cell node or a child of the cell node.
+ * @param cells - an iterable of Cells
+ * @param isCellNode - a function that takes in a node and checks if
+ * it is a cell node.
+ */
+function findCell(
+  node: HTMLElement,
+  cells: IterableOrArrayLike<Cell>,
+  isCellNode: (node: HTMLElement) => boolean
```

Comment:
This is a nice function reuse. I wonder if we should provide a default `isCellNode` that just checks for the `jp-Cell` class so that this function works well in the absence of a more specific filtering function. Not a huge deal either way, though.